components/dashboards-web-component/src/designer/DashboardDesigner.jsx

```diff
@@ -265,11 +304,82 @@ export default class DashboardDesigner extends Component {
                 pageId = this.state.dashboard.landingPage;
             }
 
-            let dashboard = this.state.dashboard;
+            let dashboard = this.getDashboard();
+
+            let optionsMap = {};
+
+            /**
+             * helper function to get the page from a given dashboard given the pageId
+             * */
+            function getPage(dashboard, pageID) {
```

Comment:
please move this function out of the functions scope of updatePageContent()

Reply:
fixed with 4a7ed67e33bafe933e1d67a67d0920516a350894